src/builder.cpp

```diff
@@ -450,93 +413,100 @@ Ref<ArrayMesh> Builder::create_mesh_from_surface(LMSurface& surf)
 	arrays[Mesh::ARRAY_INDEX] = indices;
 
 	// Create mesh
-	Ref<ArrayMesh> mesh = memnew(ArrayMesh());
 	mesh->add_surface_from_arrays(Mesh::PRIMITIVE_TRIANGLES, arrays);
-	return mesh;
 }
 
-void Builder::build_texture_mesh(int idx, const char* name, LMEntity& ent, Node3D* parent, ColliderType coltype, ColliderShape colshape)
+MeshInstance3D* Builder::build_entity_mesh(int idx, LMEntity& ent, Node3D* parent, ColliderType coltype, ColliderShape colshape)
 {
-	// Create material
-	Ref<Material> material;
+	// Create instance name based on entity idx
+	String instance_name = String("entity_{0}_geometry").format(Array::make(idx));
```

Comment:
Doing this loses the entity name given in Trenchbroom. Do we really need that?

Reply:
Actually, this might only lose the texture name, so it probably doesn't matter that much.

Reply:
Yeah the point of that is that since we are now grouping textures into one mesh its a better idea to just use the entity id